### ISSUE TITLE
Learning to Learn

### ISSUE THREAD

**Issue by sweaterr** [sweaterr/readinglist#346]
http://bair.berkeley.edu/blog/2017/07/18/learning-to-learn/

**Comment by sweaterr** [comment]
A key aspect of intelligence is versatility – the capability of doing many different things.
인텔리전스의 주요 측면은 다양한 기능을 수행 할 수있는 다 기능성입니다.

Current AI systems excel at mastering a single skill, such as Go, Jeopardy, or even helicopter aerobatics.
현재 인공 지능 시스템은 Go, Jeopardy 또는 심지어 헬리콥터 곡예와 같은 단일 스킬을 숙달하는 데 탁월합니다.

But, when you instead ask an AI system to do a variety of seemingly simple problems, it will struggle.
그러나 AI 시스템에 다양한 겉으로보기에 간단한 문제를 요구하면 어려울 것입니다.

A champion Jeopardy program cannot hold a conversation, and an expert helicopter controller for aerobatics cannot navigate in new, simple situations such as locating, navigating to, and hovering over a fire to put it out.
챔피언 Jeopardy 프로그램은 대화를 가질 수 없으며 곡예 비행을위한 전문가 헬리콥터 컨트롤러는 화재를 찾아 내고 탐색하며 공중 선회와 같이 새롭고 간단한 상황에서 탐색 할 수 없습니다.

In contrast, a human can act and adapt intelligently to a wide variety of new, unseen situations.
대조적으로 인간은 다양한 새롭고 보이지 않는 상황에 지능적으로 행동하고 적응할 수 있습니다.

How can we enable our artificial agents to acquire such versatility?.
인공 에이전트가 어떻게 그러한 다양성을 획득 할 수 있습니까?

**Comment by sweaterr** [comment]
Current AI systems can master a complex skill from scratch, using an understandably large amount of time and experience.
현재 인공 지능 시스템은 당연히 많은 시간과 경험을 사용하여 복잡한 기술을 처음부터 마스터 할 수 있습니다.

But if we want our agents to be able to acquire many skills and adapt to many environments, we cannot afford to train each skill in each setting from scratch.
그러나 대리인이 많은 기술을 습득하고 다양한 환경에 적응할 수있게하려면 각 설정에서 각 기술을 처음부터 교육 할 여력이 없습니다.

Instead, we need our agents to learn how to learn new tasks faster by reusing previous experience, rather than considering each new task in isolation.
대신 새로운 요원을 개별적으로 고려하는 대신 이전 경험을 재사용하여 새로운 작업을보다 빠르게 학습하는 방법을 배우기 위해 에이전트가 필요합니다.

This approach of learning to learn, or meta-learning, is a key stepping stone towards versatile agents that can continually learn a wide variety of tasks throughout their lifetimes.
학습이나 메타 학습을 배우는이 접근법은 평생 동안 계속해서 다양한 작업을 배울 수있는 다재다능한 에이전트를 향한 중요한 발판입니다.

**Comment by sweaterr** [comment]
## So, what is learning to learn, and what has it been used for?
Early approaches to meta-learning date back to the late 1980s and early 1990s, including Jürgen Schmidhuber’s thesis and work by Yoshua and Samy Bengio.
메타 러닝에 대한 초기 접근 방식은 1980 년대 후반과 1990 년대 초반에 Jürgen Schmidhuber의 논문과 Yoshua와 Samy Bengio의 작업을 포함하여 거슬러 올라간다.

Recently meta-learning has become a hot topic, with a flurry of recent papers, most commonly using the technique for hyperparameter and neural network optimization, finding good network architectures, few-shot image recognition, and fast reinforcement learning.
최근 meta-learning은 최근 논문의 혼란과 함께, hyperparameter 및 신경망 최적화를위한 기술, 좋은 네트워크 아키텍처 찾기, 소수 이미지 인식 및 빠른 강화 학습을 가장 일반적으로 사용하여 뜨거운 주제가되었습니다.

![image](https://user-images.githubusercontent.com/1518919/38787555-9fc27016-4169-11e8-9326-7531151cce6d.png)

**Comment by sweaterr** [comment]
## Few-Shot Learning

In 2015, Brendan Lake et al, published a paper that challenged modern machine learning methods to be able to learn new concepts from one or a few instances of that concept.
2015 년 Brendan Lake 등은 현대 기계 학습 방법에 도전하여 하나 또는 몇 가지 개념에서 새로운 개념을 배울 수있는 논문을 발표했습니다.

As an example, Lake suggested that humans can learn to identify “novel two-wheel vehicles” from a single picture (e.g, as shown on the right), whereas machines cannot generalize a concept from just a single image.
예를 들어 레이크 (Lake)는 인간이 단일 사진 (예 : 오른쪽 그림)에서 "소설의 2 륜차"를 식별하는 것을 배울 수 있지만 기계는 단 하나의 이미지만으로 개념을 일반화 할 수 없다고 제안했습니다.

(Humans can also draw a character in a new alphabet after seeing just one example).
(인간은 한 가지 예를 본 후에 새로운 알파벳으로 인물을 그릴 수도 있습니다).

Along with the paper, Lake included a dataset of handwritten characters, Omniglot, the “transpose” of MNIST, with 1623 character classes, each with 20 examples.
이 신문과 함께 레이크에는 1623 개의 문자 클래스와 20 개의 예가있는 MNIST의 "전치"인 손으로 쓴 문자 Omniglot의 데이터 세트가 포함되었습니다.

Two deep learning models quickly followed with papers at ICML 2016 that used memory-augmented neural networks and sequential generative models; showing it is possible for deep models to learn to learn from a few examples, though not yet at the level of humans.
2 가지 심층 학습 모델은 ICML 2016에서 메모리가 추가 된 신경망 및 순차 생성 모델을 사용한 논문을 빠르게 따라갔습니다. 깊은 모델이 인간의 수준에서는 아니지만 몇 가지 예를 통해 배우는 법을 배울 수 있음을 보여줍니다.

**Comment by sweaterr** [comment]
## How Recent Meta-learning Approaches Work

Meta-learning systems are trained by being exposed to a large number of tasks and are then tested in their ability to learn new tasks; an example of a task might be classifying a new image within 5 possible classes, given one example of each class, or learning to efficiently navigate a new maze with only one traversal through the maze.
메타 학습 시스템은 많은 수의 과제에 노출되어 훈련을 받고 새로운 과제를 배우는 능력을 테스트합니다. 작업의 예로는 5 개의 가능한 클래스 내에서 새 이미지를 분류하거나, 각 클래스의 예제를 하나씩 제공하거나, 미로를 통과하는 하나의 탐색만으로 새로운 미로를 효율적으로 탐색하는 방법을 학습 할 수 있습니다.

This differs from many standard machine learning techniques, which involve training on a single task and testing on held-out examples from that task.
이것은 단일 작업에 대한 교육과 그 작업에서 보류 된 예제에 대한 테스트와 관련된 많은 표준 기계 학습 기술과 다릅니다.

**Comment by sweaterr** [comment]
![image](https://user-images.githubusercontent.com/1518919/38787698-9ab8ad82-416a-11e8-8470-6f969167f5b4.png)

During meta-learning, the model is trained to learn tasks in the meta-training set.
메타 학습 중에이 모델은 메타 트레이닝 세트에서 작업을 학습하도록 훈련받습니다.

There are two optimizations at play – the learner, which learns new tasks, and the meta-learner, which trains the learner.
학습에는 새로운 작업을 배우는 학습자와 학습자를 교육하는 메타 학습자라는 두 가지 최적화가 있습니다.

Methods for meta-learning have typically fallen into one of three categories: recurrent models, metric learning, and learning optimizers.
메타 학습을위한 방법은 대개 반복 모델, 메트릭 학습 및 학습 최적화 도구의 세 가지 범주 중 하나로 분류됩니다.

**Comment by sweaterr** [comment]
## Recurrent Models
These approaches train a recurrent model, e.g, an LSTM, to take in the dataset sequentially and then process new inputs from the task., In an image classification setting, this might involve passing in the set of (image, label) pairs of a dataset sequentially, followed by new examples which must be classified.
이러한 접근법은 LSTM과 같은 반복 모델을 교육하여 데이터 세트를 순차적으로 가져 와서 작업에서 새로운 입력을 처리합니다. 이미지 분류 설정에서이 작업은 (이미지, 레이블) 쌍의 순차적으로 데이터 세트를 분류하고 분류해야하는 새로운 예제가 뒤 따릅니다.


![image](https://user-images.githubusercontent.com/1518919/38787732-cd31d5d6-416a-11e8-8d19-49aec3d51540.png)

The meta-learner uses gradient descent, whereas the learner simply rolls out the recurrent network.
메타 학습자는 그라디언트 디센트를 사용하는 반면, 학습자는 반복적 인 네트워크를 단순히 롤아웃합니다.

This approach is one of the most general approaches and has been used for few-shot classification and regression, and meta-reinforcement learning.
이 접근법은 가장 일반적인 접근법 중 하나이며 소수 샷 분류 및 회귀 및 메타 보강 학습에 사용되었습니다.

Due to its flexibility, this approach also tends to be less (meta-)efficient than other methods because the learner network needs to come up with its learning strategy from scratch.
이 방법의 유연성으로 인해 학습자 네트워크가 처음부터 학습 전략을 수립해야하기 때문에이 방법은 다른 방법보다 효율적입니다 (메타).

**Comment by sweaterr** [comment]
## Metric Learning

The meta-learner uses gradient descent, whereas the learner simply rolls out the recurrent network.
메타 학습자는 그라디언트 디센트를 사용하는 반면, 학습자는 반복적 인 네트워크를 단순히 롤아웃합니다.

This approach is one of the most general approaches and has been used for few-shot classification and regression, and meta-reinforcement learning.
이 접근법은 가장 일반적인 접근법 중 하나이며 소수 샷 분류 및 회귀 및 메타 보강 학습에 사용되었습니다.

Due to its flexibility, this approach also tends to be less (meta-)efficient than other methods because the learner network needs to come up with its learning strategy from scratch.
이 방법의 유연성으로 인해 학습자 네트워크가 처음부터 학습 전략을 수립해야하기 때문에이 방법은 다른 방법보다 효율적입니다 (메타).

**Comment by sweaterr** [comment]
## Learning Optimizers

The final approach is to learn an optimizer.
마지막 방법은 최적화 프로그램을 배우는 것입니다.

In this method, there is one network (the meta-learner) which learns to update another network (the learner) so that the learner effectively learns the task.
이 방법에는 학습자가 효과적으로 작업을 학습 할 수 있도록 다른 네트워크 (학습자)를 업데이트하는 것을 배우는 하나의 네트워크 (메타 학습자)가 있습니다.

This approach has been extensively studied for better neural network optimization.
이 접근법은 더 나은 신경망 최적화를 위해 광범위하게 연구되었습니다.

The meta-learner is typically a recurrent network so that it can remember how it previously updated the learner model.
메타 학습자는 일반적으로 반복적 인 네트워크이므로 이전에 학습자 모델을 어떻게 업데이트했는지 기억할 수 있습니다.

The meta-learner can be trained with reinforcement learning or supervised learning.
메타 학습자는 강화 학습 또는 감독 학습을 통해 학습 할 수 있습니다.

Ravi & Larochelle recently demonstrated this approach’s merit for few-shot image classification, presenting the view that the learner model is an optimization process that should be learned.
Ravi & Larochelle은 최근 소수점 이미지 분류에 대한이 접근법의 장점을 보여 주었고, 학습자 모델은 배워야하는 최적화 프로세스라는 견해를 나타 냈습니다.

**Comment by sweaterr** [comment]
## Learning Initializations as Meta-Learning

Arguably, the biggest success story of transfer learning has been initializing vision network weights using ImageNet pre-training.
대체로 전송 학습의 가장 큰 성공 사례는 ImageNet 사전 교육을 사용하여 비전 네트워크 가중치를 초기화하는 것입니다.

In particular, when approaching any new vision task, the well-known paradigm is to first collect labeled data for the task, acquire a network pre-trained on ImageNet classification, and then fine-tune the network on the collected data using gradient descent.
특히 새로운 비전 작업에 접근 할 때 잘 알려진 패러다임은 작업에 대한 레이블이 지정된 데이터를 수집하고 ImageNet 분류에 사전 교육 된 네트워크를 확보 한 다음 그라디언트 디센트를 사용하여 수집 된 데이터를 네트워크에 맞게 미세 조정하는 것입니다.

Using this approach, neural networks can more effectively learn new image-based tasks from modestly-sized datasets.
이 접근 방식을 사용하면 신경 네트워크는 적절한 크기의 데이터 세트에서 새로운 이미지 기반 작업을보다 효과적으로 학습 할 수 있습니다.

However, pre-training only goes so far.
그러나 예비 교육은 지금까지만 진행됩니다.

Because the last layers of the network still need to be heavily adapted to the new task, datasets that are too small, as in the few-shot setting, will still cause severe overfitting.
네트워크의 마지막 레이어는 여전히 새로운 작업에 많이 적응해야하기 때문에 소수 설정과 같이 너무 작은 데이터 세트는 여전히 심한 과장을 초래할 수 있습니다.

Furthermore, we unfortunately don’t have an analogous pre-training scheme for non-vision domains such as speech, language, and control.1 Is there something to learn from the remarkable success of ImageNet fine-tuning?.
또한 불행히도 언어, 언어 및 제어와 같은 비 시각적 도메인을위한 유사한 사전 교육 체계가 없습니다 .1 ImageNet 미세 조정의 놀라운 성공을 통해 배울 점이 있습니까?

**Comment by sweaterr** [comment]
## Model-Agnostic Meta-Learning (MAML)

What if we directly optimized for an initial representation that can be effectively fine-tuned from a small number of examples? This is exactly the idea behind our recently-proposed algorithm, model-agnostic meta-learning (MAML).
적은 수의 예제에서 효과적으로 미세 조정할 수있는 초기 표현을 직접 최적화했다면 어떻게 될까요? 이는 최근 제안 된 알고리즘 인 MAML (Model-Agnostic Meta-Learning)의 아이디어입니다.

Like other meta-learning methods, MAML trains over a wide range of tasks.
다른 메타 학습 방법과 마찬가지로 MAML은 다양한 작업을 수행합니다.

It trains for a representation that can be quickly adapted to a new task, via a few gradient steps.
몇 가지 그라디언트 단계를 통해 새 작업에 신속하게 적용 할 수있는 표현을위한 교육을 제공합니다.

The meta-learner seeks to find an initialization that is not only useful for adapting to various problems, but also can be adapted quickly (in a small number of steps) and efficiently (using only a few examples).
메타 러너는 다양한 문제에 적응하는 데 유용 할뿐만 아니라 신속하게 (적은 단계로) 적응할 수 있고 (몇 가지 예를 사용하여) 효율적으로 초기화 할 수 있습니다.

Below is a visualization – suppose we are seeking to find a set of parameters θ that are highly adaptable.
아래는 시각화입니다. 매우 적응력이 좋은 매개 변수 집합 θ를 찾으려고합니다.

During the course of meta-learning (the bold line), MAML optimizes for a set of parameters such that when a gradient step is taken with respect to a particular task i (the gray lines), the parameters are close to the optimal parameters θ∗i for task i.
메타 학습 (굵은 선) 과정에서 MAML은 특정 작업 i (회색 선)에 대해 기울기 단계를 수행 할 때 매개 변수가 최적 매개 변수 θ에 가까워 지도록 매개 변수 세트를 최적화합니다 * 작업 i를위한 i.

**Comment by sweaterr** [comment]
This approach is quite simple, and has a number of advantages.
이 방법은 매우 간단하며 여러 가지 장점이 있습니다.

It doesn’t make any assumptions on the form of the model.
그것은 모델의 형태에 대한 어떠한 가정도하지 않는다.

It is quite efficient – there are no additional parameters introduced for meta-learning and the learner’s strategy uses a known optimization process (gradient descent), rather than having to come up with one from scratch.
매우 효율적입니다. 메타 러닝을 위해 도입 된 추가 매개 변수가 없으며 학습자의 전략은 기존 최적화 과정 (그라디언트 디센트)을 사용합니다. 처음부터 새로 고안하지 않아도됩니다.

Lastly, it can be easily applied to a number of domains, including classification, regression, and reinforcement learning.
마지막으로 분류, 회귀 및 강화 학습을 비롯한 여러 도메인에 쉽게 적용 할 수 있습니다.

**Comment by sweaterr** [comment]
Despite the simplicity of the approach, we were surprised to find that the method was able to substantially outperform a number of existing approaches on popular few-shot image classification benchmarks, Omniglot and MiniImageNet2, including existing approaches that were much more complex or domain specific.
이 접근법의 단순성에도 불구하고 우리는이 기법이 훨씬 복잡하거나 도메인 특유의 기존 접근법을 포함하여 인기있는 소수 샷 이미지 분류 벤치 마크 인 Omniglot 및 MiniImageNet2의 기존 접근 방식보다 훨씬 뛰어난 성능을 보였다는 사실에 놀랐다.

Beyond classification, we also tried to learn how to adapt a simulated robot’s behavior to different goals, akin to the motivation at the top of this blog post – versatility.
분류를 넘어, 우리는 시뮬레이션 된 로봇의 행동을 다른 목표에 적응시키는 방법을 배우려고 노력했습니다.이 블로그 게시물의 다양성의 상단에있는 동기와 유사합니다.

To do so, we combined MAML with policy gradient methods for reinforcement learning.
그렇게하기 위해 우리는 강화 학습을 위해 정책 구배 방법과 MAML을 결합했습니다.

MAML discovered a policy which let a simulated robot adapt its locomotion direction and speed in a single gradient update.
MAML은 시뮬레이트 된 로봇이 단일 그라디언트 업데이트에서 이동 방향과 속도를 적용하도록하는 정책을 발견했습니다.

See videos below:.
아래 동영상보기 :.

**Comment by sweaterr** [comment]
![](http://bair.berkeley.edu/static/blog/maml/cheetah_direc.gif)

**Comment by sweaterr** [comment]
The generality of the method — it can be combined with any model smooth enough for gradient-based optimization — makes MAML applicable to a wide range of domains and learning objectives beyond those explored in the paper.
이 방법의 일반성은 그라디언트 기반 최적화를 위해 충분히 부드러운 모든 모델과 결합 할 수 있으므로 MAML을 다양한 범위의 도메인 및 학습 목표에 적용 할 수 있습니다.

We hope that MAML’s simple approach for effectively teaching agents to adapt to variety of scenarios will bring us one step closer towards developing versatile agents that can learn a variety of skills in real world settings.
다양한 시나리오에 효과적으로 에이전트를 가르 칠 수있는 MAML의 단순한 접근 방법을 통해 실제 환경에서 다양한 기술을 습득 할 수있는 다양한 에이전트를 개발할 수 있습니다.
